Sass too

ricardoom opened the pull request:
@aslagle I set up sass, check it out.

Commit:
added ignore rules to cover sass & css

Commit:
added sass folder for sass usages

aslagle commented on examples/alice/css/core.css:
```diff
@@ -1,75 +1,69 @@
-@charset "UTF-8";
```
Just fyi, the files in examples/alice/css are extracted from an epub file, and we won't have control over them when we use this with other books. I'd probably leave them alone but it doesn't really matter.

ricardoom replied:
so any global stylings need to happen in the root then, yeah?

aslagle replied:
Any styling changes _within the book content_ have to be done in js, the way I have it set up now. For the nav and any of the book controls, I think we should include a new css file from index.html.

Commit:
updated gitignore file so the new scss/css .sass-cache and .map files are being ignored properly

Commit:
removed the pink test/sanity check

Commit:
removed sass watching from example file

Commit:
added sass linter

Commit:
updated the  file to start using external scss/css resource  TODO: set up to compile scss/css on npm install

Commit:
added controls.scss/css files. moving the control stylings out of  tag in  into its own directory and compiled css

Commit:
fully referencing the control styles from an external resource

Commit:
added separate color file

Commit:
referencing new colors and added @import directive to bring in the colors

Commit:
now using a separate typography file

Commit:
compiled css and importing colors via control.scss

Commit:
compiled css, importing type and colors via control.scss

Commit:
importing type

Commit:
Merge branch 'sass-too' of https://github.com/NYPL-Simplified/webpub-viewer into sass-too

Commit:
colors:
added nypl reds
type: added basefont variable from NYPL-TK
moved the bg color to the .links class so it disappears along w/ the controls on tap,
changed px values to rems

ricardoom commented:
@aslagle could you take a look at this? Do you think it can be merged? I could branch off of this and work on the node-sass bit. not sure about the sequencing...

aslagle commented:
I would like to get the node-sass commands in first, if you don't mind, to make sure I can build it.

I also don't think I want the NYPL reds in here since this is a general library. If we do an NYPL-branded version I'd make another repo that uses this library but has its own colors. The NYPL Monochrome is good though.

aslagle commented on src/styles/css/controls.css:
```diff
@@ -0,0 +1,65 @@
+
+.controls-view {
+  position: fixed;
+  top: 0.125rem;
```
This is making the settings view cover the nav, so you can't click it again to close the settings, which right now is the only way to close the settings.

ricardoom replied:
ok. i fixed the issue where the settings dialog was hiding the settings link.

Commit:
fixed a layout issue w/ the settings mobile. it should now be tapable
